### PR TITLE
bump gl-native v10.11.1, common to v23.3.1

### PR DESCRIPTION
**Pull request by ank27**
## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 Mapbox welcomes participation and contributions from everyone.
 
 # main
+## Dependencies
+* Update common to v23.3.0. ([1984](https://github.com/mapbox/mapbox-maps-android/pull/1984))
```
Comment (jush) [inline CHANGELOG.md:7]: Should we add that the known issue (NPE if last location not available) from rc.1 is fixed?

Comment (ank27) [reply CHANGELOG.md:7]: Yes, will do that when I will bump gl-native and copy both native and common changelogs.

File: sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt [hunk 2]
```diff
@@ -534,6 +534,14 @@ internal class NativeMapImpl(val map: MapInterface) :
     return map.setStyleGeoJSONSourceData(sourceId, data)
   }
 
+  override fun setStyleGeoJSONSourceData(
```
Comment (yunikkk) [inline sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt:537]: Hmm the geojson with data-id be present in 10.11?

Comment (ank27) [reply sdk/src/main/java/com/mapbox/maps/NativeMapImpl.kt]: Done.

**Review by alexander-kulikovskii — APPROVED** [review]

**Review by pengdev — APPROVED** [review]

**Review by yunikkk — APPROVED** [review]

**Pull request merged**